### PR TITLE
Add allow_dust method to TxBuilder

### PR DESCRIPTION
**Pull request by terror**
We needed this for testing our wallet with dust, does this look like a reasonable feature? If so, I'll go ahead and add a test, update the changelog, etc.

### CODE REVIEW

**Comment by danielabrozzoni** [comment]
Concept ACK! I think this could be useful for other purposes as well, like creating a transaction with a 1 satoshi notification output to someone

**Comment by danielabrozzoni** [comment]
Concept ACK! Transactions with below-dust ouputs are non-standard, but still valid, so I think BDK should be able to create them. It could also be useful in some very rare cases, like for creating a notification output to an address.

File: src/wallet/tx_builder.rs [hunk 1]
```diff
@@ -560,6 +561,12 @@ impl<'a, D: BatchDatabase, Cs: CoinSelectionAlgorithm<D>, Ctx: TxBuilderContext>
         self.params.current_height = Some(height);
         self
     }
+
+    /// Set whether or not the dust limit is checked.
```
Comment (danielabrozzoni) [inline src/wallet/tx_builder.rs:565]: Maybe we could also explain here that if the dust is allowed you might end up with a transaction that is non standard

**Comment by danielabrozzoni** [comment]
The code looks good to me, can you please squash all the commits into the first one?

**Comment by danielabrozzoni** [comment]
Mh... There are still two commits, and the latest one's message is a bit weird. I'd use interactive git rebases to fix (https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) - I would reword the first commit to change the message, and drop entirely the second commit

**Comment by terror** [comment]
> Mh... There are still two commits, and the latest one's message is a bit weird. I'd use interactive git rebases to fix (https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) - I would reword the first commit to change the message, and drop entirely the second commit

@danielabrozzoni Should be good to go now. PTAL

**Review by danielabrozzoni — APPROVED** [review]
utACK 61ad455872ab78df744763c0d69d63c6683f4814

**Review by vladimirfomene — COMMENTED** [review]
Tested ACK

**Review by rajarshimaitra — COMMENTED** [review]
Review ACK.. Code looks good to me.. A small test at this stage would be really nice..

**Comment by danielabrozzoni** [comment]
@rajarshimaitra do you have in mind any other additional test? There's already a `test_allow_dust_limit`, I can't think of any other that could be useful

**Review by rajarshimaitra — APPROVED** [review]
tACK 61ad455872ab78df744763c0d69d63c6683f4814

Huh.. I have no idea how I missed the test in last pass.. Thanks @danielabrozzoni yup thats all, we don't need anything else..

This looks good to go..

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,14 +15,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Change the meaning of the `fee_amount` field inside `CoinSelectionResult`: from now on the `fee_amount` will represent only the fees asociated with the utxos in the `selected` field of `CoinSelectionResult`.
 - New `RpcBlockchain` implementation with various fixes.
 - Return balance in separate categories, namely `confirmed`, `trusted_pending`, `untrusted_pending` & `immature`.
+- Add `allow_dust` method on `TxBuilder`.
```
Comment (afilini) [inline CHANGELOG.md]: You should rebase and move this under the new unreleased section

Comment (danielabrozzoni) [reply CHANGELOG.md]: Hey @terror, sorry for the ping, but the feature freeze is tomorrow, which means that if you fix this comment by tomorrow we can merge the PR and release `allow_dust` as a part of bdk 0.22! :)

**Review by danielabrozzoni — APPROVED** [review]
ACK 257a687344815e6c2610d30d41f8885a0cc2990a

**Comment by danielabrozzoni** [comment]
Oh no, I just noticed... @terror can you sign your commit? Otherwise I can't merge. If you prefer, I can sign it with my key.

**Comment by danielabrozzoni** [comment]
tACK bfd7b2f65debe57edc5d563c071f0bfce592a8d2
